github_lines_viewed/lines-viewed.js

```diff
@@ -0,0 +1,58 @@
+function getNewRatio(){
+    const changedFiles = document.querySelectorAll('copilot-diff-entry');
+
+    let totalLinesChanged = 0;
+    let linesViewed = 0;
+
+    changedFiles.forEach(changedFile => {
+        let fileLinesChanged = Number(changedFile.querySelector('.diffstat').textContent)
+        totalLinesChanged += fileLinesChanged
+
+        let checkbox = changedFile.querySelector('input[type="checkbox"]')
+        if (checkbox.hasAttribute('checked')) {
+            linesViewed += fileLinesChanged
+        }
+    })
+
+    return `${linesViewed} / ${totalLinesChanged}`
+}
+
+function updateLinesRead(newRatio){
+    console.log(`Updating lines read: ${newRatio}`)
+    const progressBar = document.querySelector('progress-bar');
+    progressBar.setAttribute('ratio', newRatio)
+    progressBar.increment() // Creates an error, but still works
+}
+
+// I don't think this ever runs since the progressBar.increment() fails (but works)
+function textReplacement() {
+    const progressBar = document.querySelector('progress-bar');
+    const progressBarText = progressBar.children[0].innerText;
+    const newProgressBarText = progressBarText.replaceAll('file', 'line').replaceAll('files', 'lines');
```

Comment:
`const newProgressBarText = progressBarText.replaceAll('file', 'line');` 

`.replaceAll('files', 'lines');` is redundant. // as `files` string won't be existing after replacing `file` with `line`